authentication/views.py

```diff
@@ -73,15 +73,18 @@ def get(
         """
         GET endpoint for logging a user in.
         """
-        redirect_url = get_redirect_url(request)
-        if not request.user.is_anonymous:
+        login_redirect_url = get_redirect_url(request, "next")
+        signup_redirect_url = get_redirect_url(request, "signup_next")
+        if (
+            not request.user.is_anonymous
+            and not request.user.profile.completed_onboarding
+        ):
             profile = request.user.profile
-            if (
-                not profile.completed_onboarding
-                and request.GET.get("skip_onboarding", "0") == "0"
-            ):
-                params = urlencode({"next": redirect_url})
-                redirect_url = f"{settings.MITOL_NEW_USER_LOGIN_URL}?{params}"
+            if request.GET.get("skip_onboarding", "0") == "0":
+                params = urlencode({"next": signup_redirect_url})
+                login_redirect_url = f"{settings.MITOL_NEW_USER_LOGIN_URL}?{params}"
                 profile.completed_onboarding = True
                 profile.save()
-        return redirect(redirect_url)
+            else:
+                return redirect(signup_redirect_url)
```

Comment:
It's possible that `signup_next` isn't specified and I think it's reasonable to have this implemented so that it's not required. In that case, it should fall back to `next`, otherwise the default (`/app`).